<?xml version="1.0"?>
<launch>

    <!-- ===================== Front Camera ==================-->
  	<arg name="FC_camera_name" default="fcamera" />
  	<!-- video_stream_provider can be a number as a video device or a url of a video stream -->
  	<arg name="FC_video_stream_provider" default="1" />
  	<!-- set camera fps to -->
  	<arg name="FC_set_camera_fps" default="30" />
  	<!-- set buffer queue size of frame capturing to -->
  	<arg name="FC_buffer_queue_size" default="100" />
  	<!-- frames per second to query the camera for -->
  	<arg name="FC_fps" default="30" />
  	<!-- frame_id for the camera -->
  	<arg name="FC_frame_id" default="$(arg FC_camera_name)" />
  	<!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
  	To use your own fill this arg with the corresponding url, e.g.:
  	"file:///$(find your_camera_package)/config/your_camera.yaml" -->
    <arg name="FC_camera_info_url" default="" />
  	<!-- flip the image horizontally (mirror it) -->
  	<arg name="FC_flip_horizontal" default="false" />
  	<!-- flip the image vertically -->
  	<arg name="FC_flip_vertical" default="false" />
    <!-- force width and height, 0 means no forcing -->
    <arg name="FC_width" default="0"/>
    <arg name="FC_height" default="0"/>

    <!-- ===================== Back Camera ==================-->
    <arg name="BC_camera_name" default="bcamera" />
    <arg name="BC_video_stream_provider" default="2" />
    <arg name="BC_set_camera_fps" default="30" />
    <arg name="BC_buffer_queue_size" default="100" />
    <arg name="BC_fps" default="30" />
    <arg name="BC_frame_id" default="$(arg BC_camera_name)" />
    <arg name="BC_camera_info_url" default="" />
    <arg name="BC_flip_horizontal" default="false" />
    <arg name="BC_flip_vertical" default="false" />
    <arg name="BC_width" default="0"/>
    <arg name="BC_height" default="0"/>

   
   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	<group ns="$(arg FC_camera_name)">
	    <node pkg="video_stream_opencv" type="video_stream" name="$(arg FC_camera_name)_stream" output="screen"> 
	    	<remap from="camera" to="image_raw" />
	    	<param name="camera_name" type="string" value="$(arg FC_camera_name)" />
	        <param name="video_stream_provider" type="string" value="$(arg FC_video_stream_provider)" />
	        <param name="set_camera_fps" type="double" value="$(arg FC_set_camera_fps)" />
	        <param name="buffer_queue_size" type="int" value="$(arg FC_buffer_queue_size)" />
	        <param name="fps" type="double" value="$(arg FC_fps)" />
	        <param name="frame_id" type="string" value="$(arg FC_frame_id)" />
	        <param name="camera_info_url" type="string" value="$(arg FC_camera_info_url)" />
	        <param name="flip_horizontal" type="bool" value="$(arg FC_flip_horizontal)" />
	        <param name="flip_vertical" type="bool" value="$(arg FC_flip_vertical)" />
	        <param name="width" type="int" value="$(arg FC_width)" />
	        <param name="height" type="int" value="$(arg FC_height)" />
	    </node>
	</group>

      <!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
    <group ns="$(arg BC_camera_name)">
      <node pkg="video_stream_opencv" type="video_stream" name="$(arg BC_camera_name)_stream" output="screen"> 
        <remap from="camera" to="image_raw" />
        <param name="camera_name" type="string" value="$(arg BC_camera_name)" />
          <param name="video_stream_provider" type="string" value="$(arg BC_video_stream_provider)" />
          <param name="set_camera_fps" type="double" value="$(arg BC_set_camera_fps)" />
          <param name="buffer_queue_size" type="int" value="$(arg BC_buffer_queue_size)" />
          <param name="fps" type="double" value="$(arg BC_fps)" />
          <param name="frame_id" type="string" value="$(arg BC_frame_id)" />
          <param name="camera_info_url" type="string" value="$(arg BC_camera_info_url)" />
          <param name="flip_horizontal" type="bool" value="$(arg BC_flip_horizontal)" />
          <param name="flip_vertical" type="bool" value="$(arg BC_flip_vertical)" />
          <param name="width" type="int" value="$(arg BC_width)" />
          <param name="height" type="int" value="$(arg BC_height)" />
      </node>
  </group>

    <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find mini_rover)/launch/dualCam.rviz" />

</launch>
